aesd-char-driver/main.c

```diff
@@ -215,15 +225,106 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
     //Release mutex
     mutex_unlock(&aesd_device.lock);
 
+    return retval;
+}
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int mode)
+{
+    loff_t retval;
+
+    //Check filp and buf validity
+    if(!filp)
+    {
+        PDEBUG("llseek requested without a valid filp\n");
+        retval = -EINVAL;
+        return retval;
+    }
+    
+    if(mutex_lock_interruptible(&aesd_device.lock))
+    {
+        retval = -EINTR;
+        return retval;
+    }
+
+    retval = fixed_size_llseek(filp, offset, mode, aesd_device.buffer.full_size);
```

Comment:
@jordicrosmompart think about a way you could design this such that you didn't need to hold a lock during the call to `fixed_size_llseek`